core/src/apps/cardano/sign_tx.py

```diff
@@ -740,85 +958,33 @@ def _ensure_no_payment_witness(witness: CardanoTxWitnessRequest) -> None:
         raise INVALID_STAKEPOOL_REGISTRATION_TX_WITNESSES
 
 
-async def _show_witness(
+async def _show_witness_request(
     ctx: wire.Context,
     witness_path: list[int],
+    signing_mode: CardanoTxSigningMode,
 ) -> None:
-    await _fail_or_warn_if_invalid_path(
-        ctx,
-        SCHEMA_ADDRESS,
-        witness_path,
-        WITNESS_PATH_NAME,
-    )
-
-
-async def _show_change_output_staking_warnings(
-    ctx: wire.Context,
-    keychain: seed.Keychain,
-    address_parameters: CardanoAddressParametersType,
-    amount: int,
-) -> None:
-    address_type = address_parameters.address_type
-
-    if (
-        address_type == CardanoAddressType.BASE
-        and not address_parameters.staking_key_hash
-    ):
-        await _fail_or_warn_if_invalid_path(
-            ctx,
-            SCHEMA_STAKING,
-            address_parameters.address_n_staking,
-            CHANGE_OUTPUT_STAKING_PATH_NAME,
-        )
-
-    staking_use_case = staking_use_cases.get(keychain, address_parameters)
-    if staking_use_case == staking_use_cases.NO_STAKING:
-        await show_warning_tx_no_staking_info(ctx, address_type, amount)
-    elif staking_use_case == staking_use_cases.POINTER_ADDRESS:
-        # ensured in _derive_shelley_address:
-        assert address_parameters.certificate_pointer is not None
-        await show_warning_tx_pointer_address(
-            ctx,
-            address_parameters.certificate_pointer,
-            amount,
-        )
-    elif staking_use_case == staking_use_cases.MISMATCH:
-        if address_parameters.address_n_staking:
-            await show_warning_tx_different_staking_account(
-                ctx,
-                to_account_path(address_parameters.address_n_staking),
-                amount,
-            )
-        else:
-            # ensured in _validate_base_address_staking_info:
-            assert address_parameters.staking_key_hash
-            await show_warning_tx_staking_key_hash(
-                ctx,
-                address_parameters.staking_key_hash,
-                amount,
-            )
-
-
-def _should_hide_output(path: list[int]) -> bool:
-    """Return whether the output address is from a safe path, so it could be hidden."""
-    return (
-        len(path) == BIP_PATH_LENGTH
-        and path[ACCOUNT_PATH_INDEX] <= MAX_SAFE_ACCOUNT_INDEX
-        and path[-2] < 2
-        and path[-1] < MAX_SAFE_CHANGE_ADDRESS_INDEX
-    )
-
+    is_payment = SCHEMA_PAYMENT.match(witness_path)
```

Comment:
Only applies to a particular if branch, maybe move it there?
(Otherwise, one tends to uncessarily think about how `is_payment` and `is_multisig` interact.)

Reply:
https://github.com/vacuumlabs/trezor-firmware/pull/66/commits/5464dda6bd13dee7dc3c63c36c6024753e8f3c46#diff-eaac9c946526e07b3ac5c317971467dc330f41ee2cd7da91e701aa3098365bdbR1019